src/scriptworker/cot/verify.py

```diff
@@ -164,14 +164,32 @@ def is_decision(self):
         """
         return self.task_type in DECISION_TASK_TYPES
 
+    def is_scope_in_restricted_scopes(self, scope, restricted_scopes):
+        """Determine if a scope matches in a list of restricted_scopes.
+           if one of the restricted scopes ends with '*', find a partial match.
+
+        Returns:
+            String: string of matching restricted_scope, if no match ""
+        """
+        match = [""]
+        for restricted in restricted_scopes:
+            if restricted.endswith("*"):
+                if scope.startswith(restricted.rstrip("*")):
+                    match.append(restricted)
```

Comment:
I could see us just returning `restricted` in each of these, rather than building a list. I don't have a strong opinion here; in theory, we shouldn't have overlapping restricted scopes.

Alternately, matching multiple restricted scopes may be a reason to throw an error.

Do you have thoughts here?